### PR TITLE
Implementing a  option; closes #558

### LINKED ISSUE #558: I don't believe the suggestion list should wrap with arrow keys
When the auto-complete popup shows multiple items:

image

I don't believe hitting "down" past the last item should wrap to the first item.  I believe that it should close the popup and move the cursor down a line.

Evidence:

- This is how Atom works if there is only 1 item in the box
- This is how ST3 works
- It's distracting if you accidentally encounter this (I run in to this fairly often because I hit the arrow keys right as the auto-complete pops up, and it breaks my workflow.)

Atom 1.0.3, OSX 10.10.4

### PR DESCRIPTION
**Pull request by zyklus**
Adds the following option:

image

which allows the suggestion list to auto-dismiss akin to ST3 functionality.  Enabled:

enabled

Disabled:

disabled

Closes #558

### CODE REVIEW

**Comment by benogle** [comment]
Looks like this no longer merges cleanly. Can you update it?

File: spec/autocomplete-manager-integration-spec.coffee [hunk 1]
```diff
@@ -1471,6 +1473,41 @@ describe 'Autocomplete Manager', ->
...
+          autocomplete = editorView.querySelector('.autocomplete-plus')
+          expect(autocomplete).toExist()
+
+          atom.commands.dispatch(editorView, 'core:move-up')
+          advanceClock(1)
+
+          expect(editorView.querySelector('.autocomplete-plus')).not.toExist()
+
+        # Dismisses down
+        runs ->
```
Comment (benogle) [inline spec/autocomplete-manager-integration-spec.coffee]: Maybe split these out into two different specs

File: spec/autocomplete-manager-integration-spec.coffee [hunk 2]
```diff
@@ -1471,6 +1473,41 @@ describe 'Autocomplete Manager', ->
           atom.commands.dispatch(suggestionListView, 'core:move-down')
           expect(items[0]).toHaveClass('selected')
 
+      it 'dismisses instead of wrapping when wrapSuggestions is disabled', ->
```
Comment (benogle) [inline spec/autocomplete-manager-integration-spec.coffee]: Make this a describe `when wrapSuggestions is disabled`

File: lib/main.coffee [hunk 3]
```diff
@@ -106,6 +106,12 @@ module.exports =
       items:
         type: 'string'
       order: 16
+    wrapSuggestions:
+      title: 'Wrap Selections to Top & Bottom'
```
Comment (benogle) [inline lib/main.coffee]: You can omit the title here. The description takes care of the explanation

File: lib/main.coffee [hunk 4]
```diff
@@ -106,6 +106,12 @@ module.exports =
...
+      description: 'Disable this if you want the auto suggestion list to automatically dismiss when reaching the beginning or end of the list.'
```
Comment (benogle) [inline lib/main.coffee]: How about

> Disabling this will dismiss the suggestion list when arrowing past the beginning or end of the list.

**Comment by zyklus** [comment]
Updated

File: spec/autocomplete-manager-integration-spec.coffee [hunk 5]
```diff
@@ -1486,6 +1488,45 @@ describe 'Autocomplete Manager', ->
           atom.commands.dispatch(suggestionListView, 'core:move-down')
           expect(items[0]).toHaveClass('selected')
 
+    describe 'when wrapSuggestions is disabled', ->
+      it 'dismisses suggestion dialog via moving up', ->
+        atom.config.set('autocomplete-plus.wrapSuggestions', false)
+        spyOn(provider, 'getSuggestions').andCallFake ->
+          [{text: 'ab'}, {text: 'abc'}, {text: 'abcd'}]
```
Comment (benogle) [inline spec/autocomplete-manager-integration-spec.coffee]: The above 3 lines should go in a `beforeEach`

**Comment by benogle** [comment]
Also, this now needs to be rebased on master

**Comment by benogle** [comment]
Ping... Mind rebasing on master?

**Comment by nonameolsson** [comment]
What's the status on this @benogle @zyklus ?

**Comment by lee-dohm** [comment]
It appears that it has been abandoned. I'm going to leave it open for a while longer in case @zyklus wants to pick it back up, then I'll have to close it unless someone wants to take it over.

**Comment by lee-dohm** [comment]
Thanks @zyklus for the hard work you put into this. I'm going to close this for now.

**Pull request closed without merging**